libecl/src/ecl_type.c

```diff
@@ -0,0 +1,204 @@
+/*
+   Copyright (C) 2017  Statoil ASA, Norway.
+
+   The file 'ecl_type.c' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <ert/util/util.h>
+#include <ert/ecl/ecl_util.h>
+#include <ert/ecl/ecl_type.h>
+
+/*****************************************************************/
+/* The string names for the different ECLIPSE low-level
+   types.
+*/
+#define ECL_TYPE_NAME_CHAR     "CHAR"
+#define ECL_TYPE_NAME_C010     "C010"
+#define ECL_TYPE_NAME_FLOAT    "REAL"
+#define ECL_TYPE_NAME_INT      "INTE"
+#define ECL_TYPE_NAME_DOUBLE   "DOUB"
+#define ECL_TYPE_NAME_BOOL     "LOGI"
+#define ECL_TYPE_NAME_MESSAGE  "MESS"
+
+ecl_data_type * ecl_type_alloc_copy(const ecl_data_type * src_type) {
```

Comment:
As I see it the plan here is to use value semantics - with a publicly available `struct` definition. I would therefor suggest we just avoid implementing pointer specific functions like `ecl_type_alloc_copy( )` and `ecl_type_free( )`.